src/Backend/Modules/ContentBlocks/Domain/ContentBlock/ContentBlockDataTransferObject.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Backend\Modules\ContentBlocks\Domain\ContentBlock;
+
+use Backend\Core\Language\Locale;
+use Symfony\Component\Validator\Constraints as Assert;
+
+class ContentBlockDataTransferObject
+{
+    /**
+     * @var ContentBlock
+     */
+    private $contentBlockEntity;
+
+    /**
+     * @var int
+     */
+    public $id;
+
+    /**
+     * @var int
+     */
+    public $extraId;
+
+    /**
+     * @var ?int
+     */
+    public $revisionId;
+
+    /**
+     * @var string
+     *
+     * @Assert\NotBlank(message="err.FieldIsRequired")
+     */
+    public $title;
+
+    /**
+     * @var string
+     *
+     * @Assert\NotBlank(message="err.FieldIsRequired")
+     */
+    public $template = ContentBlock::DEFAULT_TEMPLATE;
+
+    /**
+     * @var string
+     *
+     * @Assert\NotBlank(message="err.FieldIsRequired")
+     */
+    public $text;
+
+    /**
+     * @var bool
+     */
+    public $isVisible = true;
+
+    /**
+     * @var Locale
+     */
+    public $locale;
+
+    /**
+     * @var int
+     */
+    public $userId;
+
+    /**
+     * @var Status
+     */
+    public $status;
+
+    public function __construct(ContentBlock $contentBlock = null)
+    {
+        $this->contentBlockEntity = $contentBlock;
+
+        if (!$this->hasExistingContentBlock()) {
+            $this->status = Status::active();
+
+            return;
+        }
+
+        $this->id = $contentBlock->getId();
+        $this->extraId = $contentBlock->getExtraId();
+        $this->isVisible = !$contentBlock->isHidden();
+        $this->title = $contentBlock->getTitle();
+        $this->text = $contentBlock->getText();
+        $this->template = $contentBlock->getTemplate();
+        $this->userId = $contentBlock->getUserId();
+        $this->locale = $contentBlock->getLocale();
+        $this->status = $contentBlock->getStatus();
+        $this->revisionId = $contentBlock->getRevisionId();
+    }
+
+    public function getContentBlockEntity(): ContentBlock
+    {
+        return $this->contentBlockEntity;
+    }
+
+    public function hasExistingContentBlock(): bool
+    {
+        return $this->contentBlockEntity instanceof ContentBlock;
+    }
+
+    public function setContentBlockEntity(ContentBlock $contentBlockEntity): void
```

Comment:
where do you use this one?

I'm not really a fan of seeing that here, if it is needed for a command maybe we should make the entity protected and add that method in the command.

Cause this can have serious side effects, you have one entity, this DTO is populated based on that and then you can overwrite the entity but the DTO won't get the values from it